### PR TITLE
remove trainable

### LINKED ISSUE #676: No detections after training on custom dataset
I'm interested in training from scratch (or rather from an efficientNet backbone) rather than fine-tuning.  After training on my dataset, the training loss values looked good, but evaluation showed no detections, and AP=0.  Even running the trained model on images from the training set gave no detections.  I've found a few other issues (#521, #499 and #68) reporting similar problems, but following the suggestions in those issues hasn't worked so far.

To try to get a better handle on what's going wrong, I'm now trying to learn a very small dataset consisting of only eight COCO images using only the person class.  My hope is that it should be easy for the model to quickly overfit this tiny dataset, and I'll be able to quickly see it learn to make detections on the training images.  However, even on this reduced dataset, I'm still getting no detections.

My training command looks like this:

```
python main.py --mode=train --training_file_pattern=tfrecord/train*.tfrecord --validation_file_pattern=tfrecord/train*.tfrecord --model_name=efficientdet-d0 --model_dir=efficientdet-d0-single_batch  --backbone_ckpt=noisy_student_efficientnet-b0  --train_batch_size=8 --num_examples_per_epoch=256 --num_epochs=500 --hparams=person_only_config.yaml
```
Note that I'm using a efficientnet backbone from here: https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet

My yaml config file specifies that I want only the person class:

```
num_classes: 2
label_map: {1: person}
```

I set num_classes to 2 following the advice in other issues.  I have also tried with num_classes=1, which gives the same results.

My training loss curves look like this:

Screen Shot 2020-08-14 at 11 16 05 AM

This certainly looks reasonable at first glance, although I might have expected them to reach (near-)zero given that the dataset literally has only 8 images, and the model certainly has sufficient capacity to simply memorize it.  It seems at least plausible that the reduction in loss represents the initial model outputting nonsense boxes (and getting penalized severely), and then slowly learning to just output nothing (and so only getting small penalties for missing the few boxes that should be there).  

[filtered_train.txt](https://github.com/google/automl/files/5075376/filtered_train.txt)

Above is my reduced annotation set (note that it's been renamed from .json to .txt so github will accept it).  It's certainly possible I've made some mistake when generating it, but it appears to my eyes to have the right annotations.  The corresponding COCO train images are:

```
000000027519.jpg  000000209468.jpg  000000276719.jpg  000000476925.jpg
000000191280.jpg  000000243600.jpg  000000386165.jpg  000000560373.jpg
```

To evaluate the trained model, I'm using this command:

```
python main.py --mode=eval --validation_file_pattern=tfrecord/train*.tfrecord --model_name=efficientdet-d0 --model_dir=efficientdet-d0-single_batch --hparams=person_only_config.yaml --val_json_file=annotations/filtered_train.json
```
Note that I'm using my training tfrecord file and training json file, rather than a separate validation file, because I want to verify that the model has learned to overfit the training set.  The output is:

```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.024
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.022
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.024
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.044
```

I'm certain I must be doing something wrong here, because it should be very easy for the model to learn such a trivial dataset, but I've run out of ideas for places to look for mistakes.  Can anyone else spot my mistake?  I'd be happy to upload additional files or output if anything would be helpful.

### PR DESCRIPTION
**Pull request by fsx950223**
Fix #676 
Only happens when moving average decay = 0 and use_keras_model = True.
Reason: utils.get_ema_variable only return trainable variables that cause WSM variables aren't restored when set is_training_bn = False.

### CODE REVIEW

**Comment by fsx950223** [comment]
I'm confused the fix is correct or I should change
https://github.com/google/automl/blob/00e45119b1fe693c67efb68d98b5c5ddf1786f5a/efficientdet/utils.py#L79
to 
```
ema_vars = tf.variables() + \
```

**Comment by fsx950223** [comment]
I also rewrite a tf2 inference module which fix the bug too.

File: efficientdet/keras/efficientdet_keras.py [hunk 1]
```diff
@@ -124,8 +124,7 @@ def _add_wsm(self, initializer):
       name = 'WSM' + ('' if i == 0 else '_' + str(i))
       self.vars.append(
           self.add_weight(
-              initializer=initializer, name=name,
-              trainable=self.is_training_bn))
```
Comment (mingxingtan) [inline efficientdet/keras/efficientdet_keras.py:128]: Oops, I believe this trainable should always be true.

**Review by mingxingtan — APPROVED** [review]
This is awesome. Thanks for the fix.
